### PR TITLE
release: add changelog for v2.5.0, bump version

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

see title

https://github.com/DisnakeDev/disnake/milestone/2

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [ ] I have formatted the code properly by running `task lint`
    - [ ] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [x] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by shiftinv** [comment]
Updated according to latest changes, should be ready now.

**Review by onerandomusername — APPROVED** [review]
lgtm

File: disnake/__init__.py [hunk 1]
```diff
@@ -73,6 +73,6 @@ class VersionInfo(NamedTuple):
     serial: int
 
 
-version_info: VersionInfo = VersionInfo(major=2, minor=5, micro=0, releaselevel="alpha", serial=0)
+version_info: VersionInfo = VersionInfo(major=2, minor=5, micro=0, releaselevel="beta", serial=1)
```
Comment (onerandomusername) [inline disnake/__init__.py:76]: final.

File: docs/whats_new.rst [hunk 2]
```diff
@@ -12,6 +12,114 @@ This page keeps a detailed human friendly rendering of what's new and changed
 in specific versions.
 
 
+.. _vp2p5p0:
+
+v2.5.0b1
```
Comment (onerandomusername) [inline docs/whats_new.rst:17]: no beta tag

File: docs/whats_new.rst [hunk 3]
```diff
@@ -12,6 +12,114 @@ This page keeps a detailed human friendly rendering of what's new and changed
...
+---------
+
+This version adds support for API v10, improves API coverage by adding support for previously missing features,
+and contains several miscellaneous enhancements and bugfixes.
+
+Regarding the message content intent:
+Note that earlier versions will continue working fine after the message content intent deadline (April 30th 2022),
```
Comment (onerandomusername) [inline docs/whats_new.rst:24]: August 31th, 2022

File: disnake/__init__.py (hunk 1 above)
Comment (shiftinv) [reply disnake/__init__.py]: e7d72e585368ce28f1900c58188c92146acef77a

File: docs/whats_new.rst (hunk 2 above)
Comment (shiftinv) [reply docs/whats_new.rst]: e7d72e585368ce28f1900c58188c92146acef77a

File: docs/whats_new.rst (hunk 3 above)
Comment (shiftinv) [reply docs/whats_new.rst]: e841e8d4ac63bd044477dca08b789082054af3c0

File: docs/whats_new.rst [hunk 4]
```diff
@@ -12,6 +12,220 @@ This page keeps a detailed human friendly rendering of what's new and changed
...
+    - Add :func:`Guild.welcome_screen`, :func:`Guild.edit_welcome_screen`
+    - Add :attr:`Invite.guild_welcome_screen`
+
+- Support ``List[str]`` and ``Dict[str, str]`` in ``option`` parameter of :class:`disnake.ui.Select` (:issue:`326`)
+- Add :func:`Guild.search_members` (:issue:`358`, :issue:`388`)
+- Add :attr:`ModalInteraction.message` (:issue:`363`, :issue:`400`)
+- Support :func:`InteractionResponse.edit_message` for modal interactions, if modal was sent in response to component interaction (:issue:`364`, :issue:`400`)
+- Support ``reason`` parameter in :func:`Message.create_thread` and :func:`Thread.delete` (:issue:`366`)
+- Add :attr:`StageInstance.guild_scheduled_event` and :attr:`StageInstance.guild_scheduled_event_id` (:issue:`394`)
+- Add :attr:`Client.session_start_limit` (:issue:`402`)
```
Comment (onerandomusername) [inline docs/whats_new.rst:129]: and SessionStartLimit

Comment (shiftinv) [reply docs/whats_new.rst]: 574b5ed1ab4209e1f11e206fc22a872793f276ba

File: docs/whats_new.rst [hunk 5]
```diff
@@ -12,6 +12,220 @@ This page keeps a detailed human friendly rendering of what's new and changed
...
+- Allow providing ``attachments=None`` to clear attachments when editing a message (:issue:`457`)
+- Add ``__repr__`` methods to interaction data types (:issue:`458`)
+- Add :func:`VoiceChannel.delete_messages`, :func:`VoiceChannel.purge`, :func:`VoiceChannel.webhooks`, :func:`VoiceChannel.create_webhook`, and improve :func:`VoiceChannel.permissions_for` (:issue:`461`)
+- Add :attr:`AppInfo.tags`, :attr:`AppInfo.install_params`, :attr:`AppInfo.custom_install_url` (:issue:`463`)
+- Add :attr:`TextChannel.last_pin_timestamp`, :attr:`DMChannel.last_pin_timestamp`, :attr:`Thread.last_pin_timestamp` (:issue:`464`)
+- Add :attr:`MessageType.auto_moderation_action` (:issue:`465`)
+- Add temporary workaround for setting API version to avoid message content intent requirement until deadline (:issue:`467`)
+- Add :attr:`Interaction.expires_at` and :attr:`Interaction.is_expired`, automatically fall back to message edit/delete if interaction expired (:issue:`469`)
+- Add ``suppress_embeds`` parameter to message send methods (:issue:`474`)
+- Add ``filename`` and ``description`` parameters to :func:`Asset.to_file` (:issue:`475`)
```
[the remark this answers is not in the document]
Comment (shiftinv) [reply docs/whats_new.rst]: 62cc160480f1ade751035a0911634ef2c14e524f

**Review by onerandomusername — APPROVED** [review]
lgtm

**Pull request merged**